### PR TITLE
feat(sdk): bump @ar.io/sdk to use paginated requests for gateways

### PR DESCRIPTION
**Pull request by dtfiedler**
The new changes use paginated requests for fetching larger items from state, include gateways.

### CODE REVIEW

File: src/data/ar-io-data-source.ts [hunk 1]
```diff
@@ -84,22 +84,27 @@ export class ArIODataSource implements ContiguousDataSource {
...
-      // Remove node wallet from peers list if it exists
-      if (this.nodeWallet !== undefined && peers[this.nodeWallet]) {
-        delete peers[this.nodeWallet];
-      }
+      const peers: Record<string, string> = {};
+      let cursor: string | undefined;
+      do {
+        const { nextCursor, items } = await this.arIO.getGateways({
+          cursor,
+        }); // TODO: better error handling if one paged request fails we may want to still update the rest
```
[the remark this answers is not in the document]
Comment (dtfiedler) [reply src/data/ar-io-data-source.ts:92]: generate the error handling code

**Comment by karlprieb** [comment]
@dtfiedler Will you update the code to handle pagination errors, or it will come in a follow up PR?

**Comment by dtfiedler** [comment]
> @dtfiedler Will you update the code to handle pagination errors, or it will come in a follow up PR?

There is an existing try/catch that will catch the first error, but the result is an empty set of gateways. We can add add additional errror handling to return gateways are successfully fetched (100 at a time by default), but we won't be able to "continue" if one fails. Do you prefer one over the other?

**Comment by karlprieb** [comment]
> There is an existing try/catch that will catch the first error, but the result is an empty set of gateways. We can add add additional errror handling to return gateways are successfully fetched (100 at a time by default), but we won't be able to "continue" if one fails, resulting in a partial result
Of gateways.
>
> Do you prefer one over the other?
I think it makes sense to return the already fetched gateways so we have at least some to fetch data from.
As we're currently not using this data source by default, this can wait :)

**Review by karlprieb — APPROVED** [review]

**Review by djwhitt — APPROVED** [review]

**Pull request merged**